pkg/workflow/spec/evaluate/evaluate_test.go

```diff
@@ -394,6 +394,26 @@ func TestEvaluate(t *testing.T) {
 				},
 			},
 		},
+		{
+			Name: "conditionals evaluation",
+			Data: `{
+				"conditions": [{"$fn.equals": [
+					{"$type": "Parameter", "name": "first"},
+					"foobar"
+				]}]
+			}`,
+			ExpectedValue: map[string]interface{}{
+				"conditions": []interface{}{testutil.JSONInvocation("equals", []interface{}{
+					testutil.JSONParameter("first"),
+					"foobar",
+				})},
+			},
+			ExpectedUnresolvable: evaluate.Unresolvable{
+				Parameters: []evaluate.UnresolvableParameter{
+					{Name: "first"},
+				},
+			},
+		},
```

Comment:
Can we add a test where the parameter is resolved too so we can verify the boolean output?